File: 30 Days of Code/Day 4: Class vs. Instance/solution.cpp
using namespace std;
#include <iostream>

class Person {
  public:
    int age;
    Person(int initialAge);
    void amIOld();
    void yearPasses();
};

Person::Person(int initialAge) {
    // Add some more code to run some checks on initialAge
    if (initialAge < 0) {
        cout << "Age is not valid, setting age to 0.\n";
        age = 0;
    } else {
        age = initialAge;
    }
}

void Person::amIOld()
{
    // Do some computations in here and print out the correct statement to the console
    string message = "You are young.";

    if (13 <= this->age && this->age < 18) {
        message = "You are a teenager.";
    } else if (18 <= this->age) {
        message = "You are old.";
    }

    cout << message << "\n";
}

void Person::yearPasses() {
    // Increment the age of the person in here
    this->age++;
}

int main() {
    int t;
    int age;
    cin >> t;
    for (int i = 0; i < t; i++) {
        cin >> age;
        Person p(age);
        p.amIOld();
        for (int j = 0; j < 3; j++) {
            p.yearPasses();
        }
        p.amIOld();

        cout << '\n';
    }

    return 0;
}

File: 30 Days of Code/Day 4: Class vs. Instance/Solution.java
import java.io.*;
import java.util.*;

public class Person {
    private int age;	
  
	public Person(int initialAge) {
  		if (initialAge < 0) {
            System.out.println("Age is not valid, setting age to 0.");
            age = 0;
        } else {
            age = initialAge;
        }
	}

	public void amIOld() {
  		// Write code determining if this person's age is old and print the correct statement:
        String message = "You are young.";
        if (13 <= this.age && this.age < 18)
            message = "You are a teenager.";
        else if (18 <= this.age)
            message = "You are old.";
        System.out.println(message);
	}

	public void yearPasses() {
  		// Increment this person's age.
        this.age++;  
	}

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int T = sc.nextInt();
		for (int i = 0; i < T; i++) {
			int age = sc.nextInt();
			Person p = new Person(age);
			p.amIOld();
			for (int j = 0; j < 3; j++) {
				p.yearPasses();
			}
			p.amIOld();
			System.out.println();
        }
		sc.close();
    }
}